src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -262,4 +240,40 @@ class CollectorService(
         .setSequence(sequence)
         .setDirection(direction)
         .build()
+
+    private fun publishCheckpoint(request: CheckpointRequestOrBuilder, checkpoint: Checkpoint, event: Event) {
+        if (!request.hasParentEventId()) {
+            if (logger.isWarnEnabled) {
+                logger.warn("Parent id missed in request {}", request.toJson())
+            }
+            return
+        }
+        val rootEvent: Event = event
+            .name("Checkpoint")
+            .type("Checkpoint")
+            .description(request.description)
+            .endTimestamp()
+            .bodyData(EventUtils.createMessageBean("Checkpoint id '${checkpoint.id}'"))
+        if (!configuration.enableCheckpointEventsPublication) {
+            rootEvent.bodyData(EventUtils.createMessageBean("Checkpoints publication is disabled. Check the component configuration to enable it"))
+        }
+        checkpoint.sessionKeyToCheckpointData.forEach { (sessionKey: SessionKey, checkpointData: CheckpointData) ->
```

Comment:
Maybe remove explicit type specifications? It's already pretty clear what types we're working with from variable names and name of the map we're iterating over